src/Main.elm

```diff
@@ -25,40 +143,511 @@ init =
+                    , labeledFieldInput
+                        (ImmediateTrashField |> getSettingsField)
+                        (Input.checkboxInput
+                            (Form.getFieldAsBool (composeSettingsFieldPath ImmediateTrashField) page.form)
+                            [ ImmediateTrashField |> composeSettingsFieldPath |> id
+                            , class "block border border-gray-300 text-gray-900 rounded-lg p-2 h-4 w-4 disabled:bg-gray-300 disabled:cursor-not-allowed outline-0"
+                            , disabled isReadonly
+                            , attributeIf isReadonly titleForInherited
+                            ]
+                            |> Html.map FormMsg
+                        )
+                        (settingsFieldToString ImmediateTrashField)
+                    ]
+                )
+            ]
+        ]
+
+
+contactDetailsView : Page -> Bool -> Html Msg
+contactDetailsView page isReadonly =
+    let
+        contactDetailsTextInput field labelText placeholder_ =
+            labeledFieldInput
+                field
+                (Input.textInput
+                    field
+                    (inputAttributes field placeholder_ isReadonly)
+                    |> Html.map FormMsg
+                )
+                labelText
+
+        getContactDetailsField fieldType =
+            Form.getFieldAsString (composeContactDetailsFieldPath fieldType) page.form
+
+        preferredContactMethodField =
+            getContactDetailsField PreferredContactMethodField
+
+        isExpanded =
+            isGroupExpanded page.isGroupExpanded ContactDetailsGroup
+
+        ( collapseButtonText, ariaExpandedAttribute ) =
+            toggleExpandButtonDetails isExpanded
+    in
+    Html.form [ class "border border-gray-200 rounded-lg px-5 pb-2 mb-5" ]
+        [ fieldset []
+            [ legend [ class "w-full flex flex-row justify-between mb-2 pt-5" ]
+                [ span [ class "text-xl md:text-2xl lg:text-3xl" ]
+                    [ text "Contact Details"
+                    , Form.getFieldAsString (groupFieldToName ContactDetailsGroup) page.form |> .error |> showIfError
+                    ]
+                , button
+                    [ class "ml-auto"
+                    , ariaExpandedAttribute
+                    , attribute "aria-controls" (groupFieldToName ContactDetailsGroup)
+                    , type_ "button"
+                    , secondaryButtonAttributes
+                    , ToggleGroupExpansion ContactDetailsGroup isExpanded |> onClick
+                    ]
+                    [ text collapseButtonText ]
+                ]
+            , viewIf isExpanded
+                (div [ id (groupFieldToName SettingsGroup) ]
+                    [ contactDetailsTextInput (getContactDetailsField CompanyNameField) "Company name" "Company name"
```

Comment:
Just a nitpick of my own preference towards the code. There is a lot of repetition here IMHO. I'd probably create a helper function and just have the input here take the field variant like 

```
contactDetailsInput CompanyNameField,
contactDetailsInput PhoneField
```

etc.